STACpopulator/extensions/thredds.py

```diff
@@ -209,6 +209,14 @@ def thredds_helper(cls, data: dict) -> dict:
         data["thredds"] = THREDDSHelper(data["data"]["access_urls"])
         return data
 
+    def create_uid(self) -> str:
+        """Return a unique ID from the server location.
+
+        For datasets with a DRS, it might might more sense to use the dataset's metadata instead.
+        """
+        location = self.data["access_urls"]["HTTPServer"].split("/fileServer/")[1]
```

Comment:
If a dataset doesn't have an "HTTPServer" access url then can we use something else instead?

Reply:
Yes, which cases should we cover here? 

I assume at least either of HTTPServer or OPeNDAP would be available.

Reply:
The problem is that thredds doesn't guarantee that any service type will be available. A thredds server could even just have their own custom service and none of the ones we expect.

A safer assumption is that there is at least one service (so at least one access_url) and then use the uuid library to create a unique ID using that access url as a seed:

```python
for key in self.data["access_urls"]:
    hash_m = hashlib.md5()
    hash_m.update(key.encode("utf-8"))
    return uuid.UUID(hash_m.hexdigest())
raise Exception("No access urls found!") # or just log an error or something
```